LAGs

logicog opened the pull request:
This adds support for Link Aggregation Group management via the CLI and the Web interface.
Extensive documentation is provided.

vDorst reviewed (commented):
Some comments.

logicog commented on httpd/page_impl.c:
```diff
@@ -237,21 +236,50 @@ void send_mirror(void)
+		if (l != 3)
+			char_to_html(',');
+	}
+	char_to_html(']');
```
Will fix.

vDorst commented:
Thanks for the changes.

Can you see the lag status in console?
I expect `lag 0` return the current status.

just a `lag` command shows way more than needed?
```
> lag
Error: lag <lag> [port]...Error: invalid hash type:
Error: invalid hash type:4
port_lag_hash_set, lag: d0, hash: 00Link aggregation group must be 0-3!
> 
```

logicog commented:
That was an oversight. I'll add proper support for showing the lag configuration in the CLI. Concentrated too much on the Web...

vDorst commented:
That is looking great.
But I found a bug in `lag` command.
When setting `laghash 0 dip sip` which working fine.
But when try to show that status with `lag show` this works too, but when making a typo `lag randomdata` it resets the laghash value.

The log
```
> lag 0 1 4 
port_lag_members_set, lag: 00, members: 0x0090
> lag show
LAG status:
 1 member ports: 1 4  (hash: 0x7e)
 2 disabled
 3 disabled
 4 disabled

> laghash 0 dip sip
port_lag_hash_set, lag: 00, hash: 18
> lag show
LAG status:
 1 member ports: 1 4  (hash: 0x18)
 2 disabled
 3 disabled
 4 disabled

> lag radmin
port_lag_members_set, lag: 42, members: 0x0000Link aggregation group must be 0-3!
> lag show
LAG status:
 1 member ports: 1 4  (hash: 0x7e)
 2 disabled
 3 disabled
 4 disabled
```

logicog commented:
I have an idea. Will fix.

logicog commented:
I think it is now fixed. For good measure I also changed all the matches for the command names to `if else`.
```
> lag 0 1 2
port_lag_members_set, lag: 00, members: 0x0030
> lag show
LAG status:
 1 member ports: 1 2  (hash: 0x7e)
 2 disabled
 3 disabled
 4 disabled

> lag
Error: lag <lag> [port]...
> lag bla
Error: lag <lag> [port]...
> 
```
Thanks for testing so thoroughly!
